c/tskit/tables.c

```diff
@@ -154,6 +157,10 @@ check_offsets(size_t num_rows, tsk_size_t *offsets, tsk_size_t length, bool chec
     int ret = TSK_ERR_BAD_OFFSET;
     size_t j;
 
+    if (offsets == NULL) {
```

Comment:
Not entirely happy with this - either refactor to put in a comment explaining why the early exit is needed.

Reply:
> It is _possible_ to test all the OOM corner cases (I've done it for [kastore](https://github.com/tskit-dev/kastore/blob/master/c/malloc_tests.c)) but it's pretty tedious.

Doesn't look _too_ bad - it feels like this should be part of CUnit. It could run your test incrementing `fail_at` and asserting that a `CU_ASSERT_EQUAL_FATAL(ret, 0);` fails each time.

Reply:
> Doesn't look too bad - it feels like this should be part of CUnit. It could run your test incrementing fail_at and asserting that a CU_ASSERT_EQUAL_FATAL(ret, 0); fails each time.

It would be good to have if you want to follow it up - it does need some special linkage wizardry to work too though. But I guess we could have a ``malloc_tests`` file? We don't actually to that much mallocing, which is a bonus.